pkg/webui/console/components/downlink-form/index.js

```diff
@@ -0,0 +1,137 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useState, useCallback } from 'react'
+import * as Yup from 'yup'
+import { defineMessages } from 'react-intl'
+
+import SubmitButton from '../../../components/submit-button'
+import RadioButton from '../../../components/radio-button'
+import Checkbox from '../../../components/checkbox'
+import Input from '../../../components/input'
+import SubmitBar from '../../../components/submit-bar'
+import toast from '../../../components/toast'
+import Form from '../../../components/form'
+import Message from '../../../lib/components/message'
+import PropTypes from '../../../lib/prop-types'
+import sharedMessages from '../../../lib/shared-messages'
+import api from '../../api'
+import { hexToBase64 } from '../../lib/bytes'
+
+const m = defineMessages({
+  insertMode: 'Insert Mode',
+  insertModeDescription: 'Messages can either replace the downlink queue or append to it',
+  replace: 'Replace',
+  push: 'Push',
+  validateFPort: 'FPort must be between 1 and 223',
+  confirmation: 'Confirmation',
+  confirmed: 'Confirmed',
+  scheduleDownlink: 'Schedule Downlink',
+  downlinkSuccess: 'Downlink was scheduled successfully',
+  payloadDescription: 'The desired payload bytes of the downlink message',
+})
+
+const validationSchema = Yup.object({
+  _mode: Yup.string()
+    .oneOf(['replace', 'push'])
+    .isRequired(),
+  f_port: Yup.number()
+    .min(1, m.validateFPort)
+    .max(223, m.validateFPort)
+    .required(sharedMessages.validateRequired),
+  confirmed: Yup.bool().required(),
+  frm_payload: Yup.string(),
+})
+
+const DownlinkForm = ({ appId, devId }) => {
+  const [error, setError] = useState('')
+  const handleSubmit = useCallback(
+    async ({ _mode, ...values }, { setSubmitting, resetForm }) => {
+      try {
+        values.frm_payload = hexToBase64(values.frm_payload)
+        await api.downlinkQueue[_mode](appId, devId, [values])
```

Comment:
If you type `01` into the `f_port` field the backend returns an error, while this should be a valid input. This happens because its sends `"01"` instead of just `1`. Please cast form values with `validationSchema.cast(...)`.